src/views/UserSettings.vue

```diff
@@ -69,23 +73,41 @@ export default {
         currency: null,  // see initUserSettingsForm
       },
       currencyList: constants.CURRENCY_LIST,
-      languageList: constants.LANGUAGE_LIST,
+      languageList: [],
+      languagesData: null,
+      languageTranslationCompletion: null,
     }
   },
+  watch:{
+    'userSettingsForm.selectedLanguage': async function () {
+      console.log("selected language: ", this.userSettingsForm.selectedLanguage)
+      if (this.userSettingsForm.selectedLanguage !== null) {
+        this.languageTranslationCompletion = await localeManager.calculateTranslationCompletion(this.userSettingsForm.selectedLanguage.code)
+        console.log("completion: ", this.languageTranslationCompletion)
+      }
+    },
 
+  },
   computed: {
     ...mapStores(useAppStore),
     formFilled() {
       return Object.values(this.userSettingsForm).every(x => !!x)
     },
   },
-  mounted() {
+  async mounted() {
```

Comment:
"async" isn't used anywhere, are you sure we need it here ?

Reply:
in line 99 I put
`    const { default: languagesData } = await import('@/i18n/data/languages.json')
`